com.dynamo.cr/com.dynamo.cr.bob/src/com/dynamo/bob/TaskBuilder.java

```diff
@@ -0,0 +1,398 @@
+        Callable<TaskResult> callableTask = () -> {
+            TaskResult result = buildTask(task, monitor);
+            return result;
+        };
+        return callableTask;
+    }
+    private Future<TaskResult> submitTask(final Task task, final IProgress monitor) {
+        return this.executorService.submit(createCallableTask(task, monitor));
+    }
+
+    private boolean compareAllSignatures(byte[] taskSignature, List<IResource> outputResources) {
+        boolean allSigsEquals = true;
+        for (IResource r : outputResources) {
+            byte[] s = state.getSignature(r.getAbsPath());
+            if (!Arrays.equals(s, taskSignature)) {
+                allSigsEquals = false;
+                break;
+            }
+        }
+        return allSigsEquals;
+    }
+
+    // deps are the task input files generated by another task not yet completed,
+    // i.e. "solve" the dependency graph
+    private boolean hasUnresolvedDependencies(Task task) {
+        Set<IResource> dependencies = getUnresolvedDependencies(task);
+        return !dependencies.isEmpty();
+    }
+    private Set<IResource> getUnresolvedDependencies(Task task) {
+        Set<IResource> dependencies = new HashSet<>(task.getInputs());
+        dependencies.retainAll(allOutputs);
+        synchronized (completedOutputs) {
+            dependencies.removeAll(completedOutputs);
+        }
+        return dependencies;
+    }
+
+
+    private boolean checkIfResourcesExist(final List<IResource> resources) {
+        // do all output files exist?
+        boolean allResourcesExists = true;
+        for (IResource r : resources) {
+            if (!r.exists()) {
+                allResourcesExists = false;
+                break;
+            }
+        }
+        return allResourcesExists;
+    }
+
+    private TaskResult buildTask(Task task, IProgress monitor) throws IOException {
+        BundleHelper.throwIfCanceled(monitor);
+
+        logger.info("Build task " + task.getName() + " " + task.getInputsString() + " -> " + task.getOutputsString());
+
+        // if (hasUnresolvedDependencies(task)) {
+        //     // postpone task. dependent input not yet generated
+        //     // logger.warning("NOT ALL DEPS " + task.getInputsString() + " -> " + task.getOutputsString());
+        //     return null;
+        // }
+
+        final List<IResource> outputResources = task.getOutputs();
+
+        boolean allOutputsExist = checkIfResourcesExist(outputResources);
+
+        // compare all task signature. current task signature between previous
+        // signature from state on disk
+        TimeProfiler.start("compare signatures");
+        TimeProfiler.addData("color", "#FFC0CB");
+        TimeProfiler.addData("main input", String.valueOf(task.input(0)));
+        final byte[] taskSignature = task.calculateSignature();
+        boolean allSigsEquals = compareAllSignatures(taskSignature, outputResources);
+        TimeProfiler.stop();
```

Comment:
this `TimeProfiler.start("compare signatures");` removed altogether with 
```
TimeProfiler.addData("color", "#FFC0CB");
TimeProfiler.addData("main input", String.valueOf(task.input(0)));
...
TimeProfiler.stop();

Reply:
Done